src/commissaire_service/storage/custodia.py

```diff
@@ -0,0 +1,197 @@
+
+
+HTTP_SOCKET_PREFIX = 'http+unix://'
+DEFAULT_SOCKET_PATH = '/var/run/custodia/custodia.sock'
+
+
+class CustodiaStoreHandler(StoreHandlerBase):
+    """
+    Handler for securely storing secrets via a local Custodia service.
+    """
+
+    # Connection should be nearly instantaneous.
+    CUSTODIA_TIMEOUT = (1.0, 5.0)  # seconds
+
+    @classmethod
+    def check_config(cls, config):
+        """
+        This store handler has no configuration checks.
+        """
+        pass
+
+    def __init__(self, config):
+        """
+        Creates a new instance of CustodiaStoreHandler.
+        :param config: Not applicable to this handler
+        :type config: None
+        """
+        super().__init__(config)
+
+        self.session = requests.Session()
+        self.session.mount(HTTP_SOCKET_PREFIX, UnixAdapter())
+        socket_path = config.get('socket_path', DEFAULT_SOCKET_PATH)
+        self.socket_url = HTTP_SOCKET_PREFIX + quote(socket_path, safe='')
+
+    def _add_auth_headers(self, headers):
+        """
+        Adds necessary authentication headers to a dictionary.
+
+        :param headers: A dictionary of HTTP headers.
+        :type headers: dict
+        """
+        headers['REMOTE_USER'] = 'commissaire'
+
+    def _build_key_container_url(self, model_instance):
+        """
+        Builds a Custodia key container URL for the given SecretModel.
+
+        :param model_instance: A SecretModel instance.
+        :type model_instance: commissaire.model.SecretModel
+        """
+        return '{}/secrets/{}/'.format(
+            self.socket_url, model_instance._key_container)
+
+    def _build_key_url(self, model_instance):
+        """
+        Builds a Custodia key URL for the given SecretModel.
+
+        :param model_instance: A SecretModel instance.
+        :type model_instance: commissaire.model.SecretModel
+        """
+        base_url = self._build_key_container_url(model_instance)
+        return base_url + model_instance.primary_key
+
+    def _save(self, model_instance):
+        """
+        Submits a serialized SecretModel string to Custodia and returns the
+        model instance.
+
+        :param model_instance: SecretModel instance to save.
+        :type model_instance: commissaire.model.SecretModel
+        :returns: The saved model instance.
+        :rtype: commissaire.model.SecretModel
+        :raises requests.HTTPError: if the request fails
+        """
+        # Create a key container for the model.  If it already exists,
+        # catch the failure and move on.  This operation should really
+        # be idempotent, but Custodia returns a 409 Conflict.
+        # (see https://github.com/latchset/custodia/issues/206)
+        try:
+            headers = {}
+            self._add_auth_headers(headers)
+            url = self._build_key_container_url(model_instance)
+
+            response = self.session.request(
+                'POST', url, headers=headers,
```

Comment:
Since ``headers`` is static why not set it when making the ``Session`` in ``__init__``?

```python
self.session = requests.Session()
self.session.headers['REMOTE_USER'] = 'commissaire'
```

From this point on you can use ``self.session.request(..., headers={..}`` to append headers to specific requests while avoiding the need to reset ``REMOTE_USER`` each time.